server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/SpanServiceElasticsearch.java

```diff
@@ -0,0 +1,196 @@
+import org.elasticsearch.action.search.SearchType;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.indices.IndexMissingException;
+import org.elasticsearch.search.SearchHit;
+import org.hawkular.apm.api.services.StoreException;
+import org.hawkular.apm.server.api.model.zipkin.Span;
+import org.hawkular.apm.server.api.services.SpanService;
+import org.hawkular.apm.server.elasticsearch.log.MsgLogger;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * @author Pavol Loffay
+ */
+public class SpanServiceElasticsearch implements SpanService {
+
+    private static final MsgLogger msgLog = MsgLogger.LOGGER;
+
+    public static final String SPAN_TYPE = "span";
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private ElasticsearchClient client = ElasticsearchClient.getSingleton();
+
+
+    @Override
+    public Span getSpan(String tenantId, String id) {
+        GetResponse response = client.getClient()
+                .prepareGet(client.getIndex(tenantId), SPAN_TYPE, id)
+                .setRouting(id)
+                .execute()
+                .actionGet();
+
+        Span span = null;
+
+        if (!response.isSourceEmpty()) {
+            try {
+                span = deserialize(response.getSourceAsString(), Span.class);
+            } catch (IOException ex) {
+                msgLog.errorFailedToParse(ex);
+            }
+        }
+
+        msgLog.tracef("Get span with id[%s] is: %s", id, span);
+
+        return span;
+    }
+
+    @Override
+    public List<Span> getChildren(String tenantId, String id) {
+
+        List<Span> spans = new ArrayList<>();
+        try {
+            final String index = client.getIndex(tenantId);
+
+            RefreshRequestBuilder refreshRequestBuilder = client.getClient()
+                    .admin()
+                    .indices()
+                    .prepareRefresh(index);
+
+            client.getClient()
+                    .admin()
+                    .indices()
+                    .refresh(refreshRequestBuilder.request())
+                    .actionGet();
+
+            QueryBuilder query = QueryBuilders.termQuery("parentId", id);
+
+            SearchRequestBuilder request = client.getClient()
+                    .prepareSearch(index)
+                    .setTypes(SPAN_TYPE)
+                    .setSearchType(SearchType.DFS_QUERY_THEN_FETCH)
+                    .setQuery(query);
+
+            SearchResponse response = request.execute()
+                    .actionGet();
+
+            for (SearchHit searchHitFields : response.getHits()) {
+                try {
+                    spans.add(deserialize(searchHitFields.getSourceAsString(), Span.class));
+                } catch (IOException ex) {
+                    msgLog.errorFailedToParse(ex);
```

Comment:
Same concern as above: is it OK to return an inconsistent result set, if one or more records fail to parse?